web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileQuarantinesBlock.php

```diff
@@ -0,0 +1,213 @@
+ *   }
+ * )
+ */
+class ProfileQuarantinesBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The DBCDK Community Service Quarantine API.
+   *
+   * @var QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * Drupal's date formatter to format dates to Drupal Date Formats.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatter $dateFormatter
+   */
+  protected $dateFormatter;
+
+  /**
+   * Creates a Profiles Block instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The DBCDK Community Service Quarantine API.
+   * @param \Drupal\Core\Datetime\DateFormatter $date_formatter
+   *   Drupal's date formatter to format dates to Drupal Date Formats.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ProfileApi $profile_api, QuarantineApi $quarantine_api, DateFormatter $date_formatter) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->profileApi = $profile_api;
+    $this->quarantineApi = $quarantine_api;
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('dbcdk_community.api.profile'),
+      $container->get('dbcdk_community.api.quarantine'),
+      $container->get('date.formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $context_username = $this->getContext('username')->getContextData()->getValue();
+    $quarantines = [];
+    try {
+      // Fetch a Community Profile based on the username so we can the
+      // corresponding ID to fetch quarantines.
+      $profile_filter = [
+        'where' => [
+          'username' => $context_username,
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Model\Profile */
+      $profile = $this->profileApi->profileFindOne(json_encode($profile_filter));
+
+      // Fetch quarantines.
+      $quarantine_filter = [
+        'order' => 'end DESC',
+        'where' => [
+          'quarantinedId' => $profile->getId(),
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Model\Quarantine[] */
+      $quarantines = $this->quarantineApi->quarantineFind(json_encode($quarantine_filter));
```

Comment:
Again: Is the typehint necessary? Also we need to cast to array to ensure handling of empty responses which are returned as `NULL`.